### ISSUE TITLE
[BUG] Propeller should reload if oauth client secret changes

### ISSUE THREAD

**Issue by sbrunk** [flyteorg/flyte#1977]
### Describe the bug

Currently, propeller does not detect and load a changed if `client_secret` in `flyte-secret-auth` until it is restarted. So if you change `client_secret` *after* propeller has been deployed, it fails to process any workflow and will log errors like this:
```
E1220 16:10:36.391892       1 workers.go:102] error syncing 'mandant1-development/f3359d6b5cd941830000': Workflow[] failed. ErrorRecordingError: failed to publish event, caused by: EventSinkError: Error sending event, caused by [rpc error: code = Unauthenticated desc = token parse error [JWT_VERIFICATION_FAILED] Could not retrieve id token from metadata, caused by: rpc error: code = Unauthenticated desc = Request unauthenticated with IDToken]
```

### Expected behavior

if `client_secret` in `flyte-secret-auth` is changed, propeller (possibly also flytescheduler) should detect the changes, reload and use the new secret.

### Additional context to reproduce

1. Deploy flyte with auth enabled but without setting `client_secret` (defaults to `foobar`)
2. Change `client_secret` in `flyte_secret_auth` to a valid secret
3. Flytepropeller will still try to use the default secret until you restart it

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by kumare3** [comment]
cc @EngHabu / @pmahindrakar-oss this seems like a fundamental problem? I actually question the need for using authN within the flyte environment

**Comment by kumare3** [comment]
> @kumare3 it was a problem in our specific setup with Keycloak because it occurred on every helm update. That's now solved via #1976 so it only happens during the first install. Restarting propeller once manually is fine and it's also documented this way in #2041
> 
> 
> 
> I think right now auth is all or nothing right? So you can't enable it only for external clients without also enabling it for communication between internal components.

Yes, one reason is because all the apis are hosted by a singular entity, where you either enable auth or you don't. In some cases it may be desirable to keep propeller to admin auth unauthenicated. But if it just works then I think having it probably makes it simpler in how you setup auth
